### PR TITLE
✨ 1138 add poke puffa functionality

### PR DESCRIPTION
**Pull request by Macludde**
I named then "Pings" instead

### CODE REVIEW

File: backend/services/core/src/datasources/Member.ts [hunk 1]
```diff
@@ -112,4 +112,138 @@ export default class MemberAPI extends dbUtils.KnexDataSource {
...
+    // Get the ping history between the two users
+    const pingHistory = await this.knex<sql.Ping>('pings')
+      .where((knex) =>
+        knex.where('from_member', currentMember.id).andWhere('to_member', memberId))
+      .orWhere((knex) => knex.where('from_member', memberId).andWhere('to_member', currentMember.id))
+      .first();
+      // If there's no ping history or if the other user sent the most recent ping,
+      // then the current user is allowed to ping the other user.
+
+    if (!pingHistory) return true;
```
Comment (Macludde) [inline backend/services/core/src/datasources/Member.ts:180]: Weird formatting of comment

```suggestion
    // If there's no ping history or if the other user sent the most recent ping,
    // then the current user is allowed to ping the other user.
    if (!pingHistory) return true;
```

File: backend/services/core/src/datasources/Member.ts [hunk 2]
```diff
@@ -112,4 +112,138 @@ export default class MemberAPI extends dbUtils.KnexDataSource {
...
+          to_sent_at: this.knex.fn.now(),
+          count: this.knex.raw('?? + 1', ['count']),
+        });
+      } else if (pingHistory.to_member === memberId
+        && (pingHistory.to_sent_at ?? new Date(0)) > pingHistory.from_sent_at) {
+        await this.knex('pings').where({ id: pingHistory.id }).update({
+          from_sent_at: this.knex.fn.now(),
+          count: this.knex.raw('?? + 1', ['count']),
+        });
+      } else {
```
Comment (Macludde) [inline backend/services/core/src/datasources/Member.ts:237]: Should use isTooRecent

File: frontend/pages/members/[id]/index.tsx [hunk 3]
```diff
@@ -57,8 +65,25 @@ export default function MemberPage() {
...
+                  onClick={async () => {
+                    await ping();
+                    refetch();
+                  }}
+                >
+                  {t('member:ping')}
+                </Button>
+              </Box>
+            </Tooltip>
+          )}
```
Comment (Macludde) [inline frontend/pages/members/[id]/index.tsx:84]: Should check "core:member:ping" access

File: frontend/pages/pings/index.tsx [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+import genGetProps from '~/functions/genGetServerSideProps';
+import { useGetPingsQuery } from '~/generated/graphql';
+import { useSetPageName } from '~/providers/PageNameProvider';
+
+export default function Pings() {
+  const { t } = useTranslation(['common', 'member']);
+  useSetPageName(t('member:pings'));
+  const { data } = useGetPingsQuery();
+  const pings = data?.pings;
+  return (
```
Comment (Macludde) [inline frontend/pages/pings/index.tsx:14]: Should check core:member:ping access

File: frontend/public/locales/sv/member.json [hunk 5]
```diff
@@ -10,5 +10,11 @@
   "editMember": "Redigera profil",
   "firstSignInDesc": "Fyll in din information nedan",
   "changeProfilePicture": "Byt profilbild",
-  "uploadNewPicture": "Ladda upp ny bild"
+  "uploadNewPicture": "Ladda upp ny bild",
+  "ping": "Pinga",
+  "pings": "Pingar",
+  "pingWaitForResponse": "Du måste vänta på ping tillbaka innan du kan pinga igen",
+  "pingedYou": "pingade dig",
```
Comment (danieladugyan) [inline frontend/public/locales/sv/member.json:17]: Saknas pingTooltip

File: frontend/public/locales/en/member.json [hunk 6]
```diff
@@ -10,5 +10,12 @@
   "editMember": "Edit profile",
   "firstSignInDesc": "Fill in your information below",
   "changeProfilePicture": "Change profile picture",
-  "uploadNewPicture": "Upload new picture"
+  "uploadNewPicture": "Upload new picture",
+  "ping": "Ping",
+  "pings": "Pings",
+  "pingWaitForResponse": "You need to get pinged back for pinging again",
```
Comment (danieladugyan) [inline frontend/public/locales/en/member.json:16]: before pinging again

File: frontend/pages/pings/index.tsx [hunk 7]
```diff
@@ -0,0 +1,24 @@
+import { Stack } from '@mui/material';
+import { useTranslation } from 'next-i18next';
+import React from 'react';
+import PingCard from '~/components/Ping/PingCard';
+import genGetProps from '~/functions/genGetServerSideProps';
+import { useGetPingsQuery } from '~/generated/graphql';
+import { useSetPageName } from '~/providers/PageNameProvider';
+
+export default function Pings() {
```
Comment (danieladugyan) [inline frontend/pages/pings/index.tsx:9]: There's no obvious way to navigate to this page. I assume you get a notification upon being poked, but if you dismiss that there's no way to find this page.

**Review by danieladugyan — CHANGES_REQUESTED** [review]

File: frontend/pages/pings/index.tsx (hunk 7 above)
Comment (Macludde) [reply frontend/pages/pings/index.tsx:9]: That's the idea

**Pull request merged**